server/src/Q3Server/Startup.cs

```diff
@@ -41,7 +44,20 @@ public void Configuration(IAppBuilder app)
                     userGetterCached,
                     groupGetterCached));
 
-            app.Use<SimpleHeaderAuthenticator>();
+            app.Use<AuthParameterChecker>();
+            app.Use<UserHeaderProcessor>();
+            app.Use<TokenQueryToHeaderConverter>();

```

Comment:
As discussed offline, this is the only bit which now needs further review. Everything else good to go.